src/compiler/crystal/tools/doc/generator.cr

```diff
@@ -226,9 +237,11 @@ class Crystal::Doc::Generator
   end
 
   def doc(context, string)
-    String.build do |io|
-      Markdown.parse string, MarkdownDocRenderer.new(context, io)
+    str = isolate_flag_lines string
```

Comment:
why create new variable? couldn't that just be `string = isolate_flag_lines string`?